tests/test_csv_exporters/test_application_round_results_export.py

```diff
@@ -0,0 +1,218 @@
+
+
+def test_application_round_results_export__reservation_unit_option_ordering(graphql):
+    application_round = ApplicationRoundFactory.create_in_status_handled()
+    application = ApplicationFactory.create_in_status_in_allocation(application_round=application_round)
+    section = ApplicationSectionFactory.create(
+        application=application,
+        suitable_time_ranges__day_of_the_week=Weekday.TUESDAY,
+    )
+
+    ReservationUnitOptionFactory.create(application_section=section, reservation_unit__name="three", preferred_order=3)
+    ReservationUnitOptionFactory.create(application_section=section, reservation_unit__name="one", preferred_order=1)
+    ReservationUnitOptionFactory.create(application_section=section, reservation_unit__name="two", preferred_order=2)
+
+    exporter = ApplicationRoundResultCSVExporter(application_round_id=application_round)
+    with mock.patch(CSV_WRITER_MOCK_PATH) as mock_writer:
+        exporter.export()
+    writes = get_writes(mock_writer)
+    assert len(writes) == 4
+
+    assert writes[1].reservation_unit_name == "one"  # type: ignore
+    assert writes[2].reservation_unit_name == "two"  # type: ignore
+    assert writes[3].reservation_unit_name == "three"  # type: ignore
+
+
+def test_application_round_results_export__allocated_slot_ordering(graphql):
+    application_round = ApplicationRoundFactory.create_in_status_handled()
+    application = ApplicationFactory.create_in_status_in_allocation(application_round=application_round)
+    section = ApplicationSectionFactory.create(
+        application=application,
+        suitable_time_ranges__day_of_the_week=Weekday.TUESDAY,
+    )
+    option_2 = ReservationUnitOptionFactory.create(
+        application_section=section, reservation_unit__name="foo", preferred_order=2
+    )
+    ReservationUnitOptionFactory.create(application_section=section, reservation_unit__name="bar", preferred_order=1)
+
+    AllocatedTimeSlotFactory.create(
+        reservation_unit_option=option_2,
+        day_of_the_week=Weekday.FRIDAY,
+        begin_time=datetime.time(10, 0),
+        end_time=datetime.time(12, 0),
+    )
+    AllocatedTimeSlotFactory.create(
+        reservation_unit_option=option_2,
+        day_of_the_week=Weekday.WEDNESDAY,
+        begin_time=datetime.time(12, 0),
+        end_time=datetime.time(14, 0),
+    )
+    AllocatedTimeSlotFactory.create(
+        reservation_unit_option=option_2,
+        day_of_the_week=Weekday.WEDNESDAY,
+        begin_time=datetime.time(18, 0),
+        end_time=datetime.time(20, 0),
+    )
+
+    exporter = ApplicationRoundResultCSVExporter(application_round_id=application_round)
+    with mock.patch(CSV_WRITER_MOCK_PATH) as mock_writer:
+        exporter.export()
+    writes = get_writes(mock_writer)
+    assert len(writes) == 5
+
+    section_row: ApplicationSectionExportRow = writes[1]  # type: ignore
+    assert section_row.section_id == str(section.id)
+    assert section_row.reservation_unit_name == "bar"
```

Comment:
The unallocated res.unit option would not be exported, since the application section does have allocations in another option, and is not completely unallocated. Cf. comment in the exported class: https://github.com/City-of-Helsinki/tilavarauspalvelu-core/pull/1157/files#r1570069055